pkg/schema/service.go

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+//go:generate mockgen -destination=mock/schema_service.go -package=mock -mock_names=Service=Service . Service
+
+import (
+	"context"
+
+	commschema "github.com/conduitio/conduit-commons/schema"
```

Comment:
Why are we renaming this to `commschema`? Other files don't do that 🤔

Btw, given the many package name clashes between conduit-commons, protocol, and conduit, did we consider a different naming approach? After thinking about this for a while, this is my proposal:
- Rename this package to `schemaregistry`. I know, not ideal, but it removes the naming conflict.
- The conduit-commons package stays as it is, simply `schema`.
- For the protocol I would introduce the naming convention of prefixing packages with `p` as in "protocol", so the package would become `pschema`. Similarly, we would rename `cplugin` to `pconnector`, to indicate it contains the protocol code for the `connector` proto files.

WDYT? If we do this, I can include the `cplugin` -> `pconnector` change in my PRs.

Reply:
> Why are we renaming this to `commschema`? Other files don't do that 🤔

`cschema` didn't read accurate to me.:) So I went with `commschema` to try indicating it comes from `commons`.

As for the name changes: yeah, it does look like we're bumping into this more frequently. I would suggest we rename it once this and Raul's PR are merged, so we don't derail the current work too much.

As for the names themselves: how about we add a `c` prefix to the `commons` packages?

Reply:
For context, I'm making some pkg changes in https://github.com/ConduitIO/conduit/pull/1630

Reply:
> As for the names themselves: how about we add a `c` prefix to the `commons` packages?

I wouldn't do that for all packages in commons. We do it for _some_ already (`csync`, `cchan`, `ccontext`) where we anticipate conflicts, but I wouldn't make it a rule for all. I'd also keep `schema` named `schema` in commons, as it can be used by processors and connectors as well, so we expose a nice API for them.

Reply:
Sounds good to me. I can make the changes once this PR has been merged and once Raul merges his changes too.

Reply:
Alright, I'll add the rename of `cplugin` -> `pconnector` after I'm done with my current work 👍